combat-trainer.lic

```diff
@@ -1052,48 +1052,47 @@ class CombatTrainer
     @stop = true
   end
 
-  def parse_args(args)
-    should_debug = args.delete('debug')
-    should_app = args.delete('app')
-    darg = args.find { |arg| arg =~ /d\d+/ }
-    args.delete(darg) if darg
-    rarg = args.find { |arg| arg =~ /r\d+/ }
-    args.delete(rarg) if rarg
-    construct = args.delete('construct')
+  def set_args
+
+    arg_definitions = [
+      [
+        { name: 'debug', regex: /debug/i, optional: true },
+        { name: 'construct', regex: /construct/i, optional: true, description: 'Construct setting to override empath no attack settings.'},
+        { name: 'dance', display:'d#', regex: /d\d+/i, optional: true, description: 'Dance threshold, d2 would keep two enemies alive.'},
+        { name: 'retreat', display:'r#', regex: /r\d+/i, optional: true, description: 'Retreat threshold, r3 would stay at range with three or more enemies'},
+      ]
+    ]
+
+    args = parse_args(arg_definitions, true)
 
-    settings = get_settings(args)
+    settings = get_settings(args.flex)
```

Comment:
There are some scripts that we just can't (without a big overhaul), like smelt-deed